proposals/bound-callable-references.md

```diff
@@ -162,12 +162,24 @@ fun test() {
 }
 ```
 
-Once the LHS is type-checked and its type is determined to be `T`, the type of the whole class literal expression is `kotlin.reflect.KClass<T'>` where `T'` is a type obtained by _substituting `T`'s arguments with star projections (`*`)_. Example:
+Once the LHS is type-checked and its type is determined to be `T`, the type of the whole class literal expression is `KClass<erase(T)>`, where `erase(T)` represents the most accurate possible type-safe representation of the type `T` and is obtained by substituting `T`'s arguments with star projections (`*`), except when the type is an array.
```

Comment:
It does matter, and we may revisit it in the future, but type-checking `arrayOfStrings::class` to `KClass<Array<*>>` would be rather unhelpful from Java user's point of view. We've briefly discussed this with @erokhins and decided (for now) to type-check `arrayOfStrings::class` to `KClass<Array<String>>`. We should discuss this in detail later anyway.